### ISSUE TITLE
Improve Dictionary Grouping for English

### ISSUE THREAD

**Issue by Unwound9** [arianneorpilla/jidoujisho#390]
First of all. Thank you for your hard work and for adding English support. Jidoujisho has been a great help to me

The problem I'm facing is that some of my dictionary definitions are being detected as separate results, which pretty much messes up my workflow since the frequencies only appear with certain dictionaries. I guess this is due to some dictionaries having the word's spelling also as the reading, whereas some have the reading field empty.

![Screenshot_20240709_212447](https://github.com/arianneorpilla/jidoujisho/assets/174216005/55a37804-b4e9-4311-9da6-76e700b3ad2a)

 Yomitan has a feature for choosing the result grouping mode (Group term-reading pairs, Group related terms, and no grouping) I reckon it would be useful to add something like that for Jidoujisho, or at least to add some kind of customization or enhancement for the way the dictionary entries are displayed.

**Comment by avc1657** [comment]
> First of all. Thank you for your hard work and for adding English support. Jidoujisho has been a great help to me
> 
> The problem I'm facing is that some of my dictionary definitions are being detected as separate results, which pretty much messes up my workflow since the frequencies only appear with certain dictionaries. I guess this is due to some dictionaries having the word's spelling also as the reading, whereas some have the reading field empty.
> 
> ![IMG-20240710-WA0001](https://private-user-images.githubusercontent.com/174216005/347565434-b2c92ab1-aeaa-4540-859a-bb5163289968.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA3Mzk1MjksIm5iZiI6MTcyMDczOTIyOSwicGF0aCI6Ii8xNzQyMTYwMDUvMzQ3NTY1NDM0LWIyYzkyYWIxLWFlYWEtNDU0MC04NTlhLWJiNTE2MzI4OTk2OC5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcxMVQyMzA3MDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wYjdkNzcyNDUxMDE2NGRjYjM1OWUzZTA1NDEwYzAzMDg0YjMyOGZiOTQ3YmU1MTM0NDMyN2M1Mzk3N2YwMzBiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.AwgySXfpj3uUlHRyTGJs_2nI9hGgbwRIOfyJkD7APGQ)
> 
> Yomitan has a feature for choosing the result grouping mode (Group term-reading pairs, Group related terms, and no grouping) I reckon it would be useful to add something like that for Jidoujisho, or at least to add some kind of customization or enhancement for the way the dictionary entries are displayed.
> 
> Edit: this seems to be related to #369

See if things work better at version 2.8.9. Version 2.9.0 is buggy.

**Comment by Unwound9** [comment]
They do, but 2.8.9 doesn't support Yomitan structured content, which is most of my dictionary collection. I guess I'll just have to wait

**Comment by avc1657** [comment]
You say most of your English dictionaries use structured content?

**Comment by Unwound9** [comment]
Yeah, I'm talking about that. How exactly did you convert the dictionaries from structured content to plain text, Do you mind sharing the script or whatever you used?

**Comment by avc1657** [comment]
To convert them you need a script and the script can vary depending on the dictionary. I just asked chat gpt to write python scripts for me.

I basically prompted something like:

Write for me a python script that runs in ./ that modifies all .json files. The script is for converting a dictionary from structured content to plain text only.

For example [paste a block of the json showing its structured structured]:



But I want it looking like this [convert the block yourself to plain text so chat gpt knows what you're talking about then paste here]:


All blocks of the json files should look like pretty much the same. I want the script to be generic, which means I want all blocks being converted to plain text, bla bla bla....... 




That's pretty much it, you just need to tell chat gpt in detail what you need. Or you can just write your own code if you want. 

I'll share here my list of dicts just so you can see you can get a very decent coverage just with plain text stuff 

![SmartSelect_20240712-084024](https://github.com/user-attachments/assets/8ec0c3ba-3cc4-4336-ac9d-d1c69aefa6bc)

**Comment by Unwound9** [comment]
You were right. ChatGPT made a script that also deletes the unnecessary entries, and I just converted a couple of dictionaries. Thank you very!
The import speed is also better in 2.8.9.

**Comment by avc1657** [comment]
Which dictionaries did you manage to convert to plain text?

**Comment by nacho00112** [comment]
could you please send the script or the complete prompt, I wasn't able to make it

**Comment by avc1657** [comment]
> could you please send the script or the complete prompt, I wasn't able to make it

What dictionaries are you trying to convert?

**Comment by nacho00112** [comment]
using the 2.9.0 preview 2 solved the problem I hope I don't find the bugs you talked about

**Comment by avc1657** [comment]
> using the 2.9.0 preview 2 solved the problem I hope I don't find the bugs you talked about

2.9.0 preview 2 solves some problems and comes bundles with a few new problems, haha. As the same says, 2.9.0 is still under pre release state, so it is expected to contain bugs.
